org/mozilla/jss/provider/javax/net/JSSContext.java

```diff
@@ -0,0 +1,87 @@
+package org.mozilla.jss.provider.javax.net;
+
+import java.security.*;
+
+import javax.net.ssl.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.mozilla.jss.CryptoManager;
+import org.mozilla.jss.provider.javax.crypto.JSSKeyManager;
+import org.mozilla.jss.provider.javax.crypto.JSSTrustManager;
+import org.mozilla.jss.pkcs11.PK11Cert;
+import org.mozilla.jss.pkcs11.PK11PrivKey;
+import org.mozilla.jss.ssl.javax.JSSEngine;
+
+public class JSSContext extends SSLContextSpi {
+    public static Logger logger = LoggerFactory.getLogger(JSSContext.class);
+
+    JSSKeyManager key_manager = null;
+    JSSTrustManager trust_manager = null;
```

Comment:
The issue is we can't use a generic `KeyManager` from a generic (i.e., non-JSS) provider: our code (and especially `JSSEngine`) requires a `CryptoManager`-provided key as it contains a pointer to a `CERTCertificate` which resides in our NSS DB (in the C code underneath all this). Only our `JSSKeyManager` provides such keys (because `PK11Cert` implements the `X509Certificate` interface and the `PK11Cert` is what holds the pointer on top of the `X509Certificate` expected as the return type).

I don't think the concept of multiple `JSSKeyManagers` make sense, right? `CryptoManager` is a singleton. We can't have multiple NSS DBs (because we haven't yet exposed that functionality). There is only one implementation of JSSKeyManager. If someone extended it, they'd probably still have to rely on our `CryptoManager`.

Reply:
Currently CryptoManager is a singleton, but this is actually a limitation of the current JSS API and there were talks to fix this in the past since NSS itself supports multiple databases. So this is not needed now, but should be considered for future improvements.

Reply:
ACK, thanks @edewata!